src/lib/users/ssh_authorized_keyring.rb

```diff
@@ -0,0 +1,213 @@
+# Copyright (c) 2016 SUSE LLC.
+#  All Rights Reserved.
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+#
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require "yast"
+require "users/ssh_authorized_keys_file"
+
+module Yast
+  module Users
+    # Read, write and store SSH authorized keys.
+    #
+    # This class manages authorized keys for each home directory in the
+    # system.
+    class SSHAuthorizedKeyring
+      include Yast::Logger
+
+      # @return [Hash<String,Array<SSHAuthorizedKey>>] Authorized keys indexed by home directory
+      attr_reader :keys
+      private :keys
+
+      # The home directory does not exist.
+      class HomeDoesNotExist < StandardError; end
+
+      # Constructor
+      def initialize
+        @keys = {}
+      end
+
+      # Add/register a keys
+      #
+      # This method does not make any change to the system. For that,
+      # see #write_keys.
+      #
+      # @param home [String] Home directory where the key will be stored
+      # @return [Array<SSHAuthorizedKey>] Registered authorized keys
+      def add_keys(home, new_keys)
+        keys[home] = new_keys
+      end
+
+      # Returns the keys for a given home directory
+      #
+      # @return [Array<SSHAuthorizedKey>] List of authorized keys
+      def [](home)
+        keys[home] || []
+      end
+
+      # Determines if the keyring is empty
+      #
+      # @return [Boolean] +true+ if it's empty; +false+ otherwise
+      def empty?
+        keys.empty?
+      end
+
+      # Read keys from a given home directory and add them to the keyring
+      #
+      # @param path [String] User's home directory
+      # @return [Boolean] +true+ if some key was found
+      def read_keys(home)
+        path = authorized_keys_path(home)
+        return false unless FileUtils::Exists(path)
+        authorized_keys = Yast::Users::SSHAuthorizedKeysFile.new(path).keys
```

Comment:
Um, some methods use the `Yast::` prefix, some do not, please unify. (I'd prefer removing it except for `FileUtils` where I'd explicitly use it to avoid possible conflict with `::FileUtils`.